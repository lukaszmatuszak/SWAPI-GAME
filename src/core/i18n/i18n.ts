import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';
import LanguageDetector from 'i18next-browser-languagedetector';

import locales from './locales/index';

i18n.use(initReactI18next)
    .use(LanguageDetector)
    .init({
        ns: ['Common'],
        defaultNS: 'Common',
        resources: locales,
        fallbackLng: 'en',
        interpolation: {
            escapeValue: false,
        },
    });
