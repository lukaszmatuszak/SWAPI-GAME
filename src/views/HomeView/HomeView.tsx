import React from 'react';

const HomeView: React.FC = () => <>Home</>;

export default HomeView;
