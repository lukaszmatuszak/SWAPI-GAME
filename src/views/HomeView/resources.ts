export interface Resource {
    iconSrc: string;
    title: string;
    iconAlt: string;
    redirect: string;
}

export const RESOURCES: Resource[] = [
    {
        iconSrc: '/vader.svg',
        title: 'CHARACTERS',
        iconAlt: 'vader',
        redirect: 'characters',
    },
    {
        iconSrc: '/falcon.svg',
        title: 'STARSHIPS',
        iconAlt: 'falcon',
        redirect: 'starships',
    },
];
