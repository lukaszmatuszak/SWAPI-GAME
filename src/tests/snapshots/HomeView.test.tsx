import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import HomeView from '../../views/HomeView/HomeView';
import '../../core/i18n/i18n';

describe('HomeView Snapshot - ', () => {
    it('renders correctly', () => {
        const wrapper = shallow(<HomeView />);

        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
