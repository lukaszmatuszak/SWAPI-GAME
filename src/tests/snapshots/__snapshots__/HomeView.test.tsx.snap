// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeView Snapshot -  renders correctly 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={12}
  >
    <Styled(MuiBox)
      pb={6}
      pt={6}
    >
      <WithStyles(ForwardRef(Typography))
        align="center"
        variant="h2"
      >
        Choose game mode
      </WithStyles(ForwardRef(Typography))>
    </Styled(MuiBox)>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    container={true}
    item={true}
    justify="center"
    spacing={2}
    xs={12}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      key="vader"
      xs={4}
    >
      <Link
        to="characters"
      >
        <ResourceTile
          resource={
            Object {
              "iconAlt": "vader",
              "iconSrc": "/vader.svg",
              "redirect": "characters",
              "title": "CHARACTERS",
            }
          }
        />
      </Link>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      key="falcon"
      xs={4}
    >
      <Link
        to="starships"
      >
        <ResourceTile
          resource={
            Object {
              "iconAlt": "falcon",
              "iconSrc": "/falcon.svg",
              "redirect": "starships",
              "title": "STARSHIPS",
            }
          }
        />
      </Link>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
